.github/workflows/update-changelog.yml

```
@@ -1,51 +1,54 @@
+        if: steps.version_info.outputs.version_changed == 'true'
+        uses: orhun/git-cliff-action@v2
         with:
           config: cliff.toml
-        env:
-          OUTPUT: CHANGELOG.md
+          args: --output CHANGELOG.md
 
-      - name: Commit changes
+      - name: Commit Changes
+        if: steps.version_info.outputs.version_changed == 'true'
         run: |
```

actionlint (1.7.4)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions for more details

(expression)